### PR TITLE
Add authoring for portable build task nupkg

### PR DESCRIPTION
**Pull request by agocke**
/cc @TyOverby

### CODE REVIEW

File: src/NuGet/Microsoft.CodeAnalysis.BuildTask.Portable.nuspec [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+                <dependency id="Microsoft.Win32.Primitives" version="4.0.1-beta-23428" />
+                <dependency id="System.Console" version="4.0.0-beta-23428" />
+                <dependency id="System.Diagnostics.Tools" version="4.0.1-beta-23428" />
+                <dependency id="System.Diagnostics.Process" version="4.1.0-beta-23428" />
+                <dependency id="System.IO.Pipes" version="4.0.0-beta-23428" />
+                <dependency id="System.Runtime" version="4.0.21-beta-23428" />
+                <dependency id="System.Runtime.InteropServices.RuntimeInformation" version="4.0.0-beta-23428" />
+                <dependency id="System.IO.FileSystem" version="4.0.1-beta-23428" />
+                <dependency id="System.IO.FileSystem.DriveInfo" version="4.0.0-beta-23428" />
+            </group>
```
Comment (Pilchie) [inline src/NuGet/Microsoft.CodeAnalysis.BuildTask.Portable.nuspec]: Making all of these as `dependency` means that any project which references this package will implicitly get references to all of these dependencies, and the assets will be chosen based on the runtime target of the project, not of the host environment being compiled.

@davidfowl - this is the place where we really need to have the host/tool graph of dependencies!

**Comment by jaredpar** [comment]
Related to #6017

**Comment by TyOverby** [comment]
What will this help us do?

**Comment by Pilchie** [comment]
I still don't understand how you actually expect to use this?

**Comment by agocke** [comment]
@Pilchie You would place it as a dependency in a dnxcore-targeted exe project and it would restore the build task and the output assemblies to the output directory, giving you the roslyn build task, the compilers, and all the dependencies in the output directory.

**Comment by Pilchie** [comment]
And also referencing them from your application :frowning: 

Also, it would restore the implementations for the framework/runtime your exe is targeting, not the one you are trying to build it on.

**Pull request merged**